src/UberStrok.Realtime.Server.Game/DeathMatchGameRoom.cs

```diff
@@ -0,0 +1,83 @@
+﻿using log4net;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UberStrok.Core.Common;
+using UberStrok.Core.Views;
+
+namespace UberStrok.Realtime.Server.Game
+{
+    class DeathMatchGameRoom : BaseGameRoom
+    {
+        private readonly static ILog s_log = LogManager.GetLogger(nameof(DeathMatchGameRoom));
+        private GameRoomDataView gameRoomDataView;
+        public Dictionary<int, int> PlayerScores;
```

Comment:
`BaseGameRoom` also already maintains a list of players in the room under `BaseGameRoom.Players` and since it is a list of `GamePeer` you can access their scores through `GamePeer.Actor.Info.Kills`.

Reply:
This would remove the score handling code in the event raisers, because it is already handled by `BaseGameRoom`.